### DISCUSSION TITLE
How to install pngquant on Windows?

### DISCUSSION THREAD

**Discussion by jeremy-feng** [squidfunk/mkdocs-material#4989] [Q&A]
As mentioned in the [documentation](https://squidfunk.github.io/mkdocs-material/setup/dependencies/image-processing/#windows_1):

> Installing [pngquant](https://pngquant.org/) on Windows is a little more involved. The [pngquant-winbuild](https://github.com/jibsen/pngquant-winbuild) repository contains a guide on how to set up an environment for building [pngquant](https://pngquant.org/) on Windows.

However, I still have no idea how to install pngquant on Windows. Nothing happened after I entered `git submodule update --init --recursive` .

I also tried to go to [pngquant.org](https://pngquant.org/) and downloaded the [Binary for Windows](https://pngquant.org/pngquant-windows.zip). But I don't konw what to do next.

**Comment by squidfunk** [comment]
Thanks for asking. I currently don't have a Windows machine at my disposal, so I can only guess. Have you tried to search the interwebs for directions? If you manage to make it work, it would be amazing if you could share it here!

**Reply by jeremy-feng** [reply]
Thanks! I will share it if I find a solution. 😉

**Reply by jeremy-feng** [reply]
I downloaded the [Binary for Windows](https://pngquant.org/pngquant-windows.zip).

Unzip it and I can get the following files:

![image](https://user-images.githubusercontent.com/44312563/216828205-03c6fc6b-8a90-4c81-9371-d6c14ad156aa.png)

Actually I can drag some png files to the `Drag PNG here to reduce palette automatically.bat` script. The images are quickly compressed. It seems that the `pngquant.exe` works well on Windows.

I think we need to figure out how to let the `optimize` plugin to tell where pngquant is. Maybe I should specify the path of pngquant.

**Reply by squidfunk** [reply]
> I think we need to figure out how to let the optimize plugin to tell where pngquant is. Maybe I should specify the path of pngquant.

Yes, that's definitely a useful configuration option. If you manage to find out how to adjust the PATH, and add the executable to a directory contained in it, it should work out of the box. However, we can certainly add an option for configuration.

**Reply by tamasgt** [reply]
Hello,

I think the pngquant-winbuild [mentioned in the documentation](https://squidfunk.github.io/mkdocs-material/plugins/requirements/image-processing/#pngquant-windows) is wrong. I don't know what it can be used for, but there's no need for it.

The solution step by step:

1) Download what you wrote: **"Binary for Windows"** from https://pngquant.org/
2) Unzip.
3) I moved it here (but wherever you want): C:\Users\USERNAME\pngquant
4) **Add this path to Environment Variables --> System variables --> Path**
5) Restart terminal or VSCode.

That's all.

```mkdocs serve```

![Snag_63052e](https://github.com/squidfunk/mkdocs-material/assets/5910815/3b5445cc-12cd-426c-b29c-655dcabaa8f3)

```mkdocs build```

![Snag_7b57ed](https://github.com/squidfunk/mkdocs-material/assets/5910815/ff44e5fc-6e86-4af3-9abf-909e525a6d87)

**Reply by alexvoss** [reply]
Thanks for this. I will have a look. Could be that we can streamline the instructions.

**Comment by jeremy-feng** [comment]
For those who want to compress png files, you can use [Pngyu](http://nukesaq88.github.io/Pngyu/). See the [repo](https://github.com/nukesaq88/Pngyu).

I found it super easy to use. Just drag your png files, or even an entire folder into it. It can compress all png files automatically, without changing the file name.

![image](https://user-images.githubusercontent.com/44312563/216829000-c913d59c-6e37-476b-8cba-a2192d13d623.png)

**Reply by squidfunk** [reply]
Thanks for sharing! Note that we plan to add support for automatically creating responsive images to the optimize plugin soon, which is an optimization technique that goes beyond compression, but for the time being, not to go through the troubles of installing pngquant, this is definitely a viable solution!
